pandas/core/series.py

```diff
@@ -2622,8 +2622,8 @@ def combine(self, other, func, fill_value=None) -> "Series":
             new_name = ops.get_op_result_name(self, other)
             new_values = []
             for idx in new_index:
-                lv = self.get(idx, fill_value)
-                rv = other.get(idx, fill_value)
+                lv = self.get_strict(idx, fill_value)
```

Comment:
So IIUC this is failing because `.get` is throwing a TypeError?

Reply:
I think it fails because .get will accept invalid keys (which are also numeric values) as index values rather than rejecting them as invalid keys.

Reply:
that makes sense.  if we're only using it here, id rather in-line get_strict